src/DataGrid.tsx

```diff
@@ -487,21 +479,29 @@ function DataGrid<R, SR, K extends Key>(
   /**
    * callbacks
    */
-  const handleColumnResize = useCallback(
+  const handleColumnResize = useLatestFunc(
```

Comment:
May be we can move the function down and add `handleColumnResizeLatest` for consistency 
https://github.com/adazzle/react-data-grid/blob/628d4e15168fd3e64dc7711a449b056dc5f1bbe4/src/DataGrid.tsx#L398